qiskit_experiments/randomized_benchmarking/rb_utils.py

```diff
@@ -0,0 +1,262 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019-2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+RB Helper functions
+"""
+
+from typing import Tuple, Dict, Optional, Iterable, List
+import numpy as np
+from qiskit import QiskitError, QuantumCircuit
+from qiskit.providers.backend import Backend
+
+
+class RBUtils:
+    """A collection of utility functions for computing additional data
+    from randomized benchmarking experiments"""
+
+    @staticmethod
+    def get_error_dict_from_backend(
```

Comment:
I think this assumes too specific situation. The backend might measure those gate errors by interleaved benchmarking, which might not give you a constant error ratio among the gates (in our backend, `sx` and `x` report exactly the same EPG thus there is assumption of error ratio based on the coherence limit, but this is not always correct if they have different gate time, or gate time is too short and gates are operated with gate time in the vicinity of 10/Δ, i.e. leakage error). In another example, if a backend has CZ and iSWAP gate, discussing error ratio of these gates doesn't make sense (we need to use IRB in this case anyways). 

So I suggest following:

* Add `gate_error_ratio` option to analysis. You need a validation that error ratio is defined for all the basis gates that are provided to the transpiler. User should be able to manually specify this ratio, such as `{"sx": 1, "x": 1.2, "rz": 0.}` -- based on their experience with the device.

* You can also keep `get_error_dict_from_backend` method as a helper function to generate above option, if not specified. The interface would be `get_error_dict_from_backend(backend, basis_gates: List[str], qubits: Iterable[int])`, and you can provide `basis_gates` from the transpiler options (if you know gate of interest, you can use `[properties().gate_error(gate, qubits) for gate in basis_gates]` to extract gate errors, just one line!). 

* You can provide above error ratio dict from the experiment class, thus you don't need to pass entire backend object, which is usually quite heavy.

Reply:
Thanks, it's similar to thoughts I had myself, I'll do as you suggested.

Reply:
Now I remember why I was reluctant to set this in the experiment class: Since the error ratios are device-dependent, and the experiment object itself is device-independent, since we can pass different backends for different runs. Still, I'll modify according to your suggestions.